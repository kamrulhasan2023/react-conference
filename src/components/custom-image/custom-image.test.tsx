import { fireEvent, render } from '@testing-library/react';
import { CustomImage, CustomFillImage } from '.';

it('Custom Image: Renders image with appropriate src', () => {
  const { getByRole } = render(
    <CustomImage
      alt="Test Alt"
      height={400}
      src="https://cpmr-islands.org/wp-content/uploads/sites/4/2019/07/Test-Logo-Small-Black-transparent-1.png"
      width={400}
    />
  );
  const element = getByRole('img');
  expect(element).toHaveAttribute(
    'src',
    'http://localhost/_next/image?url=https%3A%2F%2Fcpmr-islands.org%2Fwp-content%2Fuploads%2Fsites%2F4%2F2019%2F07%2FTest-Logo-Small-Black-transparent-1.png&w=828&q=75'
  );
  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});

it('Custom Image: Renders fallback image when src is broken', async () => {
  const { getByRole } = render(
    <CustomImage
      alt="Test Alt"
      height={400}
      src="/invalid-src-link"
      width={400}
    />
  );
  const element = getByRole('img');
  await fireEvent.error(element);
  expect(element).toHaveAttribute(
    'src',
    'http://localhost/_next/image?url=%2Ffallback-image.jpg&w=828&q=75'
  );
  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});

it('Custom Fill Image: Renders image with appropriate src', () => {
  const { getByRole } = render(
    <CustomFillImage
      alt="Test Alt"
      src="https://cpmr-islands.org/wp-content/uploads/sites/4/2019/07/Test-Logo-Small-Black-transparent-1.png"
    />
  );
  const element = getByRole('img');
  expect(element).toHaveAttribute(
    'src',
    'http://localhost/_next/image?url=https%3A%2F%2Fcpmr-islands.org%2Fwp-content%2Fuploads%2Fsites%2F4%2F2019%2F07%2FTest-Logo-Small-Black-transparent-1.png&w=3840&q=75'
  );
  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});

it('Custom Fill Image: Renders fallback image when src is broken', async () => {
  const { getByRole } = render(
    <CustomFillImage alt="Test Alt" src="/invalid-src-link" />
  );
  const element = getByRole('img');
  await fireEvent.error(element);
  expect(element).toHaveAttribute(
    'src',
    'http://localhost/_next/image?url=%2Ffallback-image.jpg&w=3840&q=75'
  );
  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});
