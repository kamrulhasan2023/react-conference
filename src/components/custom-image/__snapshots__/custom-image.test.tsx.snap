// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Fill Image: Renders fallback image when src is broken 1`] = `
<img
  alt="Test Alt"
  data-nimg="fill"
  decoding="async"
  loading="lazy"
  sizes="100vw"
  src="http://localhost/_next/image?url=%2Ffallback-image.jpg&w=3840&q=75"
  srcset="/_next/image?url=%2Ffallback-image.jpg&w=640&q=75 640w, /_next/image?url=%2Ffallback-image.jpg&w=750&q=75 750w, /_next/image?url=%2Ffallback-image.jpg&w=828&q=75 828w, /_next/image?url=%2Ffallback-image.jpg&w=1080&q=75 1080w, /_next/image?url=%2Ffallback-image.jpg&w=1200&q=75 1200w, /_next/image?url=%2Ffallback-image.jpg&w=1920&q=75 1920w, /_next/image?url=%2Ffallback-image.jpg&w=2048&q=75 2048w, /_next/image?url=%2Ffallback-image.jpg&w=3840&q=75 3840w"
  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; background-position: 50% 50%; background-repeat: no-repeat;"
/>
`;

exports[`Custom Fill Image: Renders image with appropriate src 1`] = `
<img
  alt="Test Alt"
  data-nimg="fill"
  decoding="async"
  loading="lazy"
  sizes="100vw"
  src="http://localhost/_next/image?url=https%3A%2F%2Fcpmr-islands.org%2Fwp-content%2Fuploads%2Fsites%2F4%2F2019%2F07%2FTest-Logo-Small-Black-transparent-1.png&w=3840&q=75"
  srcset="/_next/image?url=https%3A%2F%2Fcpmr-islands.org%2Fwp-content%2Fuploads%2Fsites%2F4%2F2019%2F07%2FTest-Logo-Small-Black-transparent-1.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fcpmr-islands.org%2Fwp-content%2Fuploads%2Fsites%2F4%2F2019%2F07%2FTest-Logo-Small-Black-transparent-1.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fcpmr-islands.org%2Fwp-content%2Fuploads%2Fsites%2F4%2F2019%2F07%2FTest-Logo-Small-Black-transparent-1.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fcpmr-islands.org%2Fwp-content%2Fuploads%2Fsites%2F4%2F2019%2F07%2FTest-Logo-Small-Black-transparent-1.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fcpmr-islands.org%2Fwp-content%2Fuploads%2Fsites%2F4%2F2019%2F07%2FTest-Logo-Small-Black-transparent-1.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fcpmr-islands.org%2Fwp-content%2Fuploads%2Fsites%2F4%2F2019%2F07%2FTest-Logo-Small-Black-transparent-1.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fcpmr-islands.org%2Fwp-content%2Fuploads%2Fsites%2F4%2F2019%2F07%2FTest-Logo-Small-Black-transparent-1.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fcpmr-islands.org%2Fwp-content%2Fuploads%2Fsites%2F4%2F2019%2F07%2FTest-Logo-Small-Black-transparent-1.png&w=3840&q=75 3840w"
  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
/>
`;

exports[`Custom Image: Renders fallback image when src is broken 1`] = `
<img
  alt="Test Alt"
  data-nimg="1"
  decoding="async"
  height="400"
  loading="lazy"
  src="http://localhost/_next/image?url=%2Ffallback-image.jpg&w=828&q=75"
  srcset="/_next/image?url=%2Ffallback-image.jpg&w=640&q=75 1x, /_next/image?url=%2Ffallback-image.jpg&w=828&q=75 2x"
  style="background-position: 50% 50%; background-repeat: no-repeat;"
  width="400"
/>
`;

exports[`Custom Image: Renders image with appropriate src 1`] = `
<img
  alt="Test Alt"
  data-nimg="1"
  decoding="async"
  height="400"
  loading="lazy"
  src="http://localhost/_next/image?url=https%3A%2F%2Fcpmr-islands.org%2Fwp-content%2Fuploads%2Fsites%2F4%2F2019%2F07%2FTest-Logo-Small-Black-transparent-1.png&w=828&q=75"
  srcset="/_next/image?url=https%3A%2F%2Fcpmr-islands.org%2Fwp-content%2Fuploads%2Fsites%2F4%2F2019%2F07%2FTest-Logo-Small-Black-transparent-1.png&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fcpmr-islands.org%2Fwp-content%2Fuploads%2Fsites%2F4%2F2019%2F07%2FTest-Logo-Small-Black-transparent-1.png&w=828&q=75 2x"
  style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
  width="400"
/>
`;
